editor/gui/editor_file_dialog.cpp

```diff
@@ -30,2661 +30,83 @@
-}
-
-void EditorFileDialog::_multi_selected(int p_item, bool p_selected) {
-	int current = p_item;
-	if (current < 0 || current >= item_list->get_item_count()) {
-		return;
-	}
-
-	Dictionary d = item_list->get_item_metadata(current);
-
-	if (!d["dir"] && p_selected) {
-		file->set_text(d["name"]);
-		_request_single_thumbnail(get_current_dir().path_join(get_current_file()));
-	}
-
-	get_ok_button()->set_disabled(_is_open_should_be_disabled());
-}
-
-void EditorFileDialog::_items_clear_selection(const Vector2 &p_pos, MouseButton p_mouse_button_index) {
-	if (p_mouse_button_index != MouseButton::LEFT) {
-		return;
-	}
-
-	item_list->deselect_all();
-
-	// If nothing is selected, then block Open button.
-	switch (mode) {
-		case FILE_MODE_OPEN_FILE:
-		case FILE_MODE_OPEN_FILES:
-			set_ok_button_text(TTRC("Open"));
-			get_ok_button()->set_disabled(!item_list->is_anything_selected());
-			break;
-
-		case FILE_MODE_OPEN_ANY:
-		case FILE_MODE_OPEN_DIR:
-			file->set_text("");
-			get_ok_button()->set_disabled(false);
-			set_ok_button_text(TTRC("Select Current Folder"));
-			break;
-
-		case FILE_MODE_SAVE_FILE:
-			// FIXME: Implement, or refactor to avoid duplication with set_mode
-			break;
-	}
-}
-
-void EditorFileDialog::_push_history() {
-	local_history.resize(local_history_pos + 1);
-	String new_path = dir_access->get_current_dir();
-	if (local_history.is_empty() || new_path != local_history[local_history_pos]) {
-		local_history.push_back(new_path);
-		local_history_pos++;
-		dir_prev->set_disabled(local_history_pos == 0);
-		dir_next->set_disabled(true);
-	}
-}
-
-void EditorFileDialog::_item_dc_selected(int p_item) {
-	int current = p_item;
-	if (current < 0 || current >= item_list->get_item_count()) {
-		return;
-	}
-
-	Dictionary d = item_list->get_item_metadata(current);
-
-	if (d["dir"]) {
-		dir_access->change_dir(d["name"]);
-		callable_mp(this, &EditorFileDialog::update_file_list).call_deferred();
-		callable_mp(this, &EditorFileDialog::update_dir).call_deferred();
-
-		_push_history();
-
-	} else {
-		_action_pressed();
-	}
-}
-
-void EditorFileDialog::_item_list_item_rmb_clicked(int p_item, const Vector2 &p_pos, MouseButton p_mouse_button_index) {
-	if (p_mouse_button_index != MouseButton::RIGHT) {
-		return;
-	}
-
-	// Right click on specific file(s) or folder(s).
-	item_menu->clear();
-	item_menu->reset_size();
-
-	// Allow specific actions only on one item.
-	bool single_item_selected = item_list->get_selected_items().size() == 1;
-
-	// Disallow deleting the .import folder, Godot kills a cat if you do and it is possibly a senseless novice action.
-	bool allow_delete = true;
-	for (int i = 0; i < item_list->get_item_count(); i++) {
-		if (!item_list->is_selected(i)) {
-			continue;
-		}
-		Dictionary item_meta = item_list->get_item_metadata(i);
-		if (String(item_meta["path"]).begins_with(ProjectSettings::get_singleton()->get_project_data_path())) {
-			allow_delete = false;
```

Comment:
This check no longer exists. I could add it as a part of ITEM_MENU_DELETE, which means the action would fail after it is clicked, instead of being entirely hidden.